addons/point_of_sale/static/src/app/store/select_lot_popup/select_lot_popup.js

```diff
@@ -91,10 +105,76 @@ export class EditListPopup extends Component {
             this.state.array.findIndex((item) => item._id == itemId),
             1
         );
+        this.state.remainingOptions = this.getRemainingOptions;
+    }
+    get getRemainingOptions() {
+        const usedValues = new Set(this.state.array.map((e) => e.text));
+        return this.props.options.filter((o) => !usedValues.has(o));
+    }
+    shouldShowOptionsForItem(item) {
+        return (
+            item._id === this.state.selectedItemId &&
+            (this.state.scrolledWithSelectedItemId !== item._id ||
+                this.state.scrolledWithSelectedItemValue !== item.text) &&
+            (!item.text ||
+                !this.props.customInput ||
+                !this.hasValidValue(item._id, item.text) ||
+                !this.props.options.includes(item.text))
+        );
+    }
+    hasValidValue(itemId, text) {
+        return (
+            (this.props.customInput || this.props.options.includes(text)) &&
+            (!this.props.uniqueValues ||
+                !this.state.array.some((elem) => elem._id !== itemId && elem.text === text))
+        );
     }
     onInputChange(itemId, text) {
         const item = this.state.array.find((elem) => elem._id === itemId);
         item.text = text;
+        this.state.remainingOptions = this.getRemainingOptions;
+        this.resetScrollInfo();
+    }
+    onScroll() {
+        const listEl = this.editListRef.el;
+        if (!listEl) {
+            return;
+        }
+        const curScrollPos = listEl.scrollTop;
+        if (
+            this.lastScrollPos &&
+            curScrollPos !== this.lastScrollPos &&
+            this.state.selectedItemId
+        ) {
+            this.state.scrolledWithSelectedItemId = this.state.selectedItemId;
+            const item = this.state.array.find((elem) => elem._id === this.state.selectedItemId);
+            this.state.scrolledWithSelectedItemValue = item.text;
+        }
+        this.lastScrollPos = curScrollPos;
+    }
+    resetScrollInfo() {
+        this.state.scrolledWithSelectedItemId = null;
+        this.state.scrolledWithSelectedItemValue = null;
+        this.lastScrollPos = null;
+    }
+    onSelectItem(itemId) {
+        this.state.selectedItemId = itemId;
+        this.resetScrollInfo();
+        if (this.toUnselectItemTimeoutId) {
+            clearTimeout(this.toUnselectItemTimeoutId);
+        }
+    }
+    onUnselectItem(itemId) {
+        // The unselection must be delayed to allow the user to click on
+        // an option of the selected item, otherwise the option disappears
+        // before the user click, and the user clicks on nothing.
+        this.toUnselectItemTimeoutId = setTimeout(() => {
+            if (this.state.selectedItemId === itemId) {
+                this.state.selectedItemId = null;
+                this.toUnselectItemTimeoutId = null;
+                this.resetScrollInfo();
+            }
+        }, 300);
```

Comment:
This timeout might not be needed. Test again.

Reply:
It is needed when clicking outside of the input and options dropdown to hide the options dropdown. Executing `this.state.selectedItemId = null;` without timeout makes clicking on an option of the dropdown not working, because the dropdown is hidden before the click is processed, making the user click on nothing instead of the clicked option.